pyxona/core.py

```diff
@@ -598,6 +598,27 @@ def _read_tracking2(self):
         # stack all coordinates into an array
         all_pos = np.concatenate([np.stack(xy, axis=-1) for xy in pos_filtered], axis=1)
 
+        # Remove NaN values at the endpoints
+        for c in range(all_pos.shape[1]):
+            r = all_pos.shape[0] - 1
```

Comment:
This value doesn't change from one iteration to another so should be defined outside of the for loop.